eclipse-collections/src/main/java/org/eclipse/collections/impl/bimap/mutable/SynchronizedBiMap.java

```diff
@@ -302,6 +303,16 @@ public V forcePut(K key, V value)
         }
     }
 
+    @Override
+    public MutableBiMap<K, V> withMapIterable(MapIterable<? extends K, ? extends V> mapIterable)
```

Comment:
Shouldn't this be in `AbstractSynchronizedMapIterable` ?

Reply:
Thanks for your comments.
Are you expecting this method to be like this below?
public MutableBiMap<K, V> withMapIterable(AbstractSynchronizedMapIterable <? extends K, ? extends V> mapIterable)
I used MapIterable instead of AbstractSynchronizedMapIterable as it is overriding default method from MutableMap. Did I miss something here? Please correct me if I am wrong.

Reply:
Sorry, let me try to rephrase it. If you check where withMap() method is implemented, this method should also be implemented in the same class. I would have expected this method to be added in the abstract class. But there might be something I might be missing, so please take a look.

Reply:
Thanks for reviewing! 
SynchronizedSortedMap, SynchronizedMutableMap and SynchronizedBiMap, all have custom implementations of withMap() method, so I decided to add withMapIterable() to each of these three classes. Also, AbstractSynchronizedMapIterable, the parent class, is inheriting withMap() method from MutableMapIterable but not overriding in the abstract class itself. In MutableMapIterable, we do have withMap() implemented so I added withMapIterable() to the class. Hope it makes sense. Thanks again!

Reply:
The actual reason why you cannot move this implementation to the abstract class is because of the covariant return type. So, you have to add it in the implementation itself.